module/zfs/vdev.c

```diff
@@ -2988,10 +2986,19 @@ vdev_dtl_load(vdev_t *vd)
 			return (error);
 		ASSERT(vd->vdev_dtl_sm != NULL);
 
-		mutex_enter(&vd->vdev_dtl_lock);
-		error = space_map_load(vd->vdev_dtl_sm,
-		    vd->vdev_dtl[DTL_MISSING], SM_ALLOC);
-		mutex_exit(&vd->vdev_dtl_lock);
+		rt = range_tree_create(NULL, RANGE_SEG64, NULL, 0, 0);
+		error = space_map_load(vd->vdev_dtl_sm, rt, SM_ALLOC);
+		if (error == 0) {
+			mutex_enter(&vd->vdev_dtl_lock);
+			rt_free = vd->vdev_dtl[DTL_MISSING];
+			vd->vdev_dtl[DTL_MISSING] = rt;
+			mutex_exit(&vd->vdev_dtl_lock);
+		} else {
+			rt_free = rt;
+		}
+
+		range_tree_vacate(rt_free, NULL, NULL);
+		range_tree_destroy(rt_free);
```

Comment:
Could there be entries in the old DTL_MISSING range_tree?  (seems so, based on the changes to vdev_stat_update) If so, do we need to merge them with with new one?  If not, we could use `range_tree_swap()` to slightly simplify this code.

Reply:
Good point.  Normally it should be empty, but it is possible it may contain new entries as a result of the import IO.  I've updated the PR to perform a merge instead of a swap.